examples/http.glu

```diff
@@ -123,8 +123,8 @@ let catch_error action catch : Handler a -> (String -> Handler a) -> Handler a =
 /// Takes a `Handler` and a `Request` tries to process the request
 let handle handler state : Handler Response -> HttpState -> IO Response =
     let not_found _ _ =
-        pure { status = NotFound, body = "Page not found" }
-    handler (\response _ -> pure response) not_found state
+        wrap { status = NotFound, body = "Page not found" }
+    handler (\response _ -> wrap response) not_found state
```

Comment:
This looks a bit clearer now in my opinion! Thoughts?

Reply:
Would this really make things easier? As it stands, I can imagine it causing problems for people looking at docs from other languages (Haskell and Idris, in my case). I can't really see it making things *easier* for anyone.

Reply:
I dunno - the name 'pure' is pretty impenetrable for most people outside the Haskell world. I added some more documentation here, with references to the haskell names https://github.com/gluon-lang/gluon/pull/319/commits/f8357afb325cf5b07011e5d6e7e8d87d3be53e74

Reply:
In the original paper, [Applicative Programming with Effects](http://www.staff.city.ac.uk/~ross/papers/Applicative.pdf), the `pure` function is described as follows:

> The idea is that `pure` embeds pure computations into the pure fragment of an effectful world—the resulting computations may thus be shunted around freely, as long as the order of the genuinely effectful computations is preserved.

But it is still very weird to call the function `pure` because it's kind of doing the opposite - wrapping a pure value to convert it into some impure thing...

Reply:
Ok, feeling better about `wrap` than `pure` at least and I haven't been able to find any other alternatives either.

While `pure` will be easier for people coming from Haskell I'd argue it is more intuitive for everyone else and anyone coming from Haskell will know to look in the `Applicative` type for the real name. Possibly we could provide `pure` as an alternative name for the same function in what is currently `make_Applicative` (or with implicits later on) though I'd be vary about providing to many such renamings.